src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SeparateSidewalkTagCheck.java

```diff
@@ -0,0 +1,309 @@
+    private static final List<String> FALLBACK_INSTRUCTIONS = Collections
+            .singletonList(INSTRUCTION_FORMAT);
+    private static final Double SIDEWALK_SEARCH_DISTANCE_DEFAULT = 15.0;
+    private static final Double EDGE_LENGTH_DEFAULT = 20.0;
+    private static final Range<Double> HEADING_DEGREE_RANGE = Ranges.closed(-20.0, 20.0);
+    private static final String MAXIMUM_HIGHWAY_DEFAULT = HighwayTag.PRIMARY.toString();
+    private final Distance searchDistance;
+    private final Distance defaultEdgeLength;
+    private final HighwayTag maximumHighwayType;
+
+    private static final String separatelyMappedTagValue = "separate";
+
+    public SeparateSidewalkTagCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.searchDistance = configurationValue(configuration, "sidewalk.search.distance",
+                SIDEWALK_SEARCH_DISTANCE_DEFAULT, Distance::meters);
+        this.defaultEdgeLength = configurationValue(configuration, "edge.length",
+                EDGE_LENGTH_DEFAULT, Distance::meters);
+        this.maximumHighwayType = this.configurationValue(configuration, "maximum.highway.type",
+                MAXIMUM_HIGHWAY_DEFAULT, str -> Enum.valueOf(HighwayTag.class, str.toUpperCase()));
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge && !this.isFlagged(object.getOsmIdentifier())
+                && ((Edge) object).isMainEdge() && HighwayTag.isCarNavigableHighway(object)
+                && this.validSidewalkFilter((Edge) object) && !((Edge) object).isClosed();
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Edge edge = (Edge) object;
+        final Location edgeMidPoint = edge.start().getLocation().midPoint(edge.end().getLocation());
+        final String sidewalkTagValue = this.getSeparateSidewalkTagValue(edge);
+        final Set<Edge> separatedSidewalks = Iterables.stream(edge.getAtlas()
+                .edgesIntersecting(this.closestSegmentToPoint(edge.asPolyLine(), edgeMidPoint)
+                        .middle().boxAround(this.searchDistance)))
+                .filter(this::validFootwayFilter).collectToSet();
+
+        if (!separatedSidewalks.isEmpty())
+        {
+            for (final Edge sidewalk : separatedSidewalks)
+            {
+                final Segment closestSidewalkSegment = this
+                        .closestSegmentToPoint(sidewalk.asPolyLine(), edgeMidPoint);
+
+                if (this.isCrossing(edge.asPolyLine(), sidewalk.asPolyLine())
+                        || !LayerTag.areOnSameLayer(edge, sidewalk))
+                {
+                    continue;
+                }
+
+                if ("right".equals(sidewalkTagValue)
+                        && !this.isRightOf(edge.asPolyLine(), closestSidewalkSegment.middle())
```

Comment:
I think I would change this logic a tiny bit so that:
1) you call the isRightOf function once for each sidewalk and keep a count of lefts and rights. (if isRightOf then right_count++ else left_count++ )
2) then (outside of the loop) you check for all three conditions.
    "right".equals(sidewalkTagValue) && right_count > 0
    "left".equals(sidewalkTagValue) && left_count > 0
    "both".equals(sidewalkTagValue) && right_count > 0 && left_count > 0

The reason for this change is so that you don't miss flagging the situation where there are two sidewalks detected on the left and the "both" tag is set. This should be flagged but your logic below would not flag it since there are two sidewalks detected nearby. Also that would prevent you from calling isRightOf multiple times per sidewalk.